### PR TITLE
Use api/v7 path in proto files

### PR DESCRIPTION
**Pull request by Joerger**
`protoc` isn't aware of go modules, so for v2+ package paths, it reads the `/vX` version suffix as an actual directory. In order to get around this, this PR passes the vendored api package in to protoc, which uses the version suffix as a directory.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -549,7 +549,7 @@ enter:
 
 # grpc generates GRPC stubs from service definitions
 .PHONY: grpc
-grpc:
+grpc: update-vendor-api
 	make -C build.assets grpc
```
Comment (andrejtokarcik) [inline Makefile:553]:
```suggestion
	$(MAKE) -C build.assets grpc
```

File: Makefile [hunk 2]
```diff
@@ -737,16 +737,21 @@ init-submodules-e: init-webapps-submodules-e
...
+	make update-vendor-api
+
+# update-vendor-api runs go mod tidy and creates a symlink between 
+# /vendor/github.com/gravitational/teleport/api/v7 and /api.
+# This is used to avoid vendoring the local api submodule.
+.PHONY: update-vendor-api
+update-vendor-api:
+	# update modules in api/
+	cd api && go mod tidy
 	rm -r vendor/github.com/gravitational/teleport/api/v7
```
Comment (andrejtokarcik) [inline Makefile:752]: If the directory is not absolutely guaranteed to exist the following would be safer:
```suggestion
	rm -rf vendor/github.com/gravitational/teleport/api/v7
```

**Review by andrejtokarcik — APPROVED** [review]

**Review by andrejtokarcik — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (Joerger) [reply Makefile]: What's the reason for changing this?

Comment (andrejtokarcik) [reply Makefile]: It's [recommended for recursive invocations](https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html) and also allows you to override the make command if needed.

**Comment by Joerger** [comment]
We're rolling back the version-path update until right before release, so this will be implemented in https://github.com/gravitational/teleport/pull/7598

**Pull request closed without merging**